llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -2321,6 +2321,29 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     }
   }
 
+  // vpcompress depends on various AVX512 extensions.
+  if (Subtarget.hasAVX512()) {
```

Comment:
Move these into the `if (!Subtarget.useSoftFloat() && Subtarget.hasAVX512())` block earlier